### PR TITLE
Update for compatibility with GWT 2.7+

### PR DESCRIPTION
**Pull request by JamesXNelson**
Only a small API change for Gwt support.

I also updated the version numbers, and made it simple to change which Gwt fork is used.  I have it defaulting to canonical, but I can swap it as needed in case I need a different fork.  I can remove that change if you don't want it to pollute master.

### CODE REVIEW

**Comment by JamesXNelson** [comment]
@niloc132 Ping for review.

File: javaee-websocket-gwt-rpc-sample/pom.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
 	<build>
 		<plugins>
 			<plugin>
-				<groupId>org.codehaus.mojo</groupId>
```
Comment (JamesXNelson) [inline javaee-websocket-gwt-rpc-sample/pom.xml]: oops

File: javaee-websocket-gwt-rpc-sample/pom.xml [hunk 2]
```diff
@@ -20,13 +20,13 @@
 		</dependency>
 
 		<dependency>
-			<groupId>com.google.gwt</groupId>
+			<groupId>${gwt.groupId}</groupId>
```
Comment (niloc132) [inline javaee-websocket-gwt-rpc-sample/pom.xml]: what, you think this'll change?

File: javaee-websocket-gwt-rpc-sample/pom.xml [hunk 3]
```diff
@@ -36,7 +36,7 @@
 	<build>
 		<plugins>
 			<plugin>
-				<groupId>org.codehaus.mojo</groupId>
+				<groupId>${gwt.plugin.groupId}</groupId>
```
Comment (niloc132) [inline javaee-websocket-gwt-rpc-sample/pom.xml]: ..okay, this miiight change, but the whole plugin block has to be rewritten if it does.

File: javaee-websocket-gwt-rpc-sample/pom.xml (hunk 2 above)
Comment (niloc132) [inline javaee-websocket-gwt-rpc-sample/pom.xml]: Oh. I get it now. Derp.

**Pull request merged**